src/test/java/com/microsoft/jenkins/acs/integration/DCOSDeploymentIT.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.jenkins.acs.integration;
+
+import com.microsoft.azure.management.compute.ContainerServiceOchestratorTypes;
+import com.microsoft.jenkins.acs.ACSDeploymentBuilder;
+import com.microsoft.jenkins.acs.ACSDeploymentContext;
+import hudson.model.Result;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.docker.commons.credentials.DockerRegistryEndpoint;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static com.microsoft.jenkins.acs.integration.TestHelpers.loadFile;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+public class DCOSDeploymentIT extends IntegrationTest {
+    @ClassRule
+    public static ACSRule acs = new ACSRule(ContainerServiceOchestratorTypes.DCOS.toString());
+
+    private ACSDeploymentContext deploymentContext;
+
+    @Before
+    @Override
+    public void setup() throws Exception {
+        super.setup();
+        setupJenkinsCredentials(acs);
+
+        deploymentContext = new ACSDeploymentContext(
+                azureCredentialsId,
+                acs.resourceGroup,
+                acs.resourceName + " | DCOS",
+                sshCredentialsId,
+                "dcos/*.json");
+    }
+
+    @Test
+    public void testSingleDirectDeployment() throws Exception {
+        loadFile(getClass(), workspace, "dcos/direct.json");
+
+        new ACSDeploymentBuilder(deploymentContext).perform(run, workspace, launcher, taskListener);
+
+        verify(run, never()).setResult(Result.FAILURE);
```

Comment:
What does this verify do?

Reply:
If we found any issue in the `ACSDeploymentBuilder` process, we will call `run.setResult(FAILURE)` to mark the build as failed.

This line checks that the build completed successfully.

Reply:
So this only check build is completed? Shall we add more code to check whether the container is really deployed?

Reply:
We have some more checks for the Kubernetes cluster, as we can get the detailed state from the K8s API. However, that's not trivial for DCOS/Swarm.

I will try to add some code to check the HTTP liveness instead.